### PR TITLE
bump bazel to 2.2.0

### PR DESCRIPTION
**Pull request by mikedanese**
/kind cleanup

```docs
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
```
E1012 02:07:20.528251       7 main.go:52] Error: Failed making a parser: unable to add directory "k8s.io/apimachinery/pkg/util/sets/types": unable to import "k8s.io/apimachinery/pkg/util/sets/types": cannot find package "k8s.io/apimachinery/pkg/util/sets/types" in any of:
	external/go_sdk/src/k8s.io/apimachinery/pkg/util/sets/types (from $GOROOT)
	bazel-out/k8-fastbuild/bin/staging/src/k8s.io/apimachinery/pkg/util/sets/gopath/src/k8s.io/apimachinery/pkg/util/sets/types (from $GOPATH)
```

**Comment by mikedanese** [comment]
Can't repro. Do you know where I can get my hands on prow's bazelrc?

**Comment by BenTheElder** [comment]
.bazelrc should be in the repo, and .bazelversion too (for bazelisk), are you using 0.23.X?

**Comment by mikedanese** [comment]
yup, this is broken on 0.23. I tried on 0.26 and 1.0.0 and everything built fine. It's probably time to bump the version.

**Comment by BenTheElder** [comment]
//:push-build is broken on newer versions. I've fixed build for 0.28.1
previously but didn't get that fixed.

**Comment by BenTheElder** [comment]
you can bazel build //... on newer versions, but CI will explode on
push-build

**Comment by fejta** [comment]
/retest
Seems to still fail all kinds of stuff?

**Comment by mikedanese** [comment]
@fejta this is blocked on a bazel version bump:

https://github.com/kubernetes/test-infra/pull/16175

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -56,6 +49,21 @@ go_register_toolchains(
     go_version = "1.13.8",
 )
 
+http_archive(
+    name = "io_k8s_repo_infra",
+    sha256 = "bf413789002d7c0946e7abca15f327300b9410ba656ccb5a8b778f70bb020d48",
+    strip_prefix = "repo-infra-962a60da343bb899e17fd974ddf49354ce527a65",
+    urls = mirror("https://github.com/mikedanese/repo-infra/archive/962a60da343bb899e17fd974ddf49354ce527a65.tar.gz"),
```
Comment (mikedanese) [inline build/root/WORKSPACE]: TODO: point at real repo-infra

**Comment by mikedanese** [comment]
Canary tests are green:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/83821/pull-kubernetes-bazel-build-canary/1242293481200685058
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/83821/pull-kubernetes-bazel-test-canary/1242293481200685057

**Pull request closed without merging**